src/main/kotlin/com/teamdev/jxbrowser/gradle/JxBrowserExtension.kt

```diff
@@ -0,0 +1,162 @@
+    public val core: Provider<String> = project.providers.provider { "$GROUP:jxbrowser:$version" }
+
+    /**
+     * Returns a dependency notation for the `jxbrowser-javafx`,
+     * an artifact with JavaFX integration.
+     */
+    public val javafx: Provider<String> = artifact("javafx")
+
+    /**
+     * Returns a dependency notation for the `jxbrowser-swing`,
+     * an artifact with Swing integration.
+     */
+    public val swing: Provider<String> = artifact("swing")
+
+    /**
+     * Returns a dependency notation for the `jxbrowser-swt`,
+     * an artifact with SWT integration.
+     */
+    public val swt: Provider<String> = artifact("swt")
+
+    /**
+     * Returns a dependency notation for the `jxbrowser-win64`,
+     * an artifact with Windows 64-bit binaries.
+     */
+    public val win64: Provider<String> = artifact("win64")
+
+    /**
+     * Returns a dependency notation for the `jxbrowser-win32`,
+     * an artifact with Windows 32-bit binaries.
+     */
+    public val win32: Provider<String> = artifact("win32")
+
+    /**
+     * Returns a dependency notation for the `jxbrowser-linux64`,
+     * an artifact with Linux 64-bit binaries.
+     */
+    public val linux64: Provider<String> = artifact("linux64")
+
+    /**
+     * Returns a dependency notation for the `jxbrowser-linux64-arm`,
+     * an artifact with Linux 64-bit ARM binaries.
+     */
+    public val linuxArm: Provider<String> = artifact("linux64-arm")
+
+    /**
+     * Returns a dependency notation for the `jxbrowser-mac`,
+     * an artifact with macOS binaries.
+     */
+    public val mac: Provider<String> = artifact("mac")
+
+    /**
+     * Returns a dependency notation for the `jxbrowser-mac-arm`,
+     * an artifact with macOS ARM binaries.
+     */
+    public val macArm: Provider<String> = artifact("mac-arm")
+
+    /**
+     * Returns a dependency notation for the `jxbrowser-cross-platform`,
+     * an artifact that includes binaries for all supported platforms.
+     */
+    public val crossPlatform: Provider<String> = artifact("cross-platform")
+
+    /**
+     * Detects the current platform and returns a dependency notation
+     * for the corresponding module with binaries.
+     */
+    public val currentPlatform: Provider<String> = currentPlatform()
+
+    private fun currentPlatform(): Provider<String> {
+        val platformMap = mapOf(
+            Pair({ isX64Bit() && isWindows() }, win64),
+            Pair({ isX64Bit() && isLinux() }, linux64),
+            Pair({ isX64Bit() && isMac() }, mac),
+            Pair({ is32Bit() && isWindows() }, win32),
+            Pair({ isArm() && isLinux() }, linuxArm),
+            Pair({ isArm() && isMac() }, macArm)
+        )
+        return platformMap.entries.firstOrNull { it.key() }?.value
+            ?: run {
+                val currentPlatform = "${osName()} ${jvmArch()}"
+                val errorMessage = "The current $currentPlatform platform is not supported by JxBrowser $version."
```

Comment:
Nit: The error messages can provide the list of supported platforms to compare the current one with the supported platforms.
```java
val errorMessage = "The current $currentPlatform platform is not supported by JxBrowser $version. The following platforms are supported: ..."
```